packages/pwa-kit-react-sdk/src/ssr/universal/components/route-component/index.js

```diff
@@ -405,9 +405,22 @@ export const routeComponent = (Wrapped, isPage, locals) => {
  */
 export const getAllRoutes = async (locals = {}) => {
     const {applicationExtensions = []} = locals
-    const extensionRoutes = (
-        await Promise.all(applicationExtensions.map((extension) => extension.getRoutes()))
-    ).flat()
+
+    let extensionRoutes = []
+    if (isServerSide){
+        extensionRoutes = (
+            await Promise.all(applicationExtensions.map((extension) => extension.getRoutes()))
+        ).flat()
+    } else {
+        // Deserialize the routes from the server
+        const serializedRoutes = window.__CONFIG__.app.routes
+        for (const applicationExtension of applicationExtensions) {
+            const extensionName = applicationExtension.constructor.name
+            const componentMap = await applicationExtension.getComponentMap()
+            const deserializedRoutes = applicationExtension.deserialize(serializedRoutes[extensionName], componentMap)
+            extensionRoutes = [...extensionRoutes, ...deserializedRoutes]
+        }
+    }
```

Comment:
I'm not a huge fan of this conditional logic here. We should investigate how we can have a single called to each getRoutes. 

We might be able to accomplish this by moving some of the deserialization logic either into the constructor of the extension, or somewhere else like that handlebar template so during the instantiation process we also deserialize. I'm not sure which one is the best just yet.